src/Type/Generic/TemplateMixedType.php

```diff
@@ -13,25 +14,40 @@
 final class TemplateMixedType extends MixedType implements TemplateType
 {
 
+	/** @var TemplateTypeScope */
+	private $scope;
+
 	/** @var string */
 	private $name;
 
+	/** @var TemplateTypeStrategy */
+	private $strategy;
+
 	public function __construct(
+		TemplateTypeScope $scope,
 		string $name,
 		bool $isExplicitMixed = false,
-		?Type $subtractedType = null
+		?Type $subtractedType = null,
+		?TemplateTypeStrategy $templateTypeStrategy = null
```

Comment:
I really try to avoid optional parameters like this. What about moving it after `$scope` and making it non-nullable?

Reply:
I made it optional such that the template type has the responsibility of choosing its default strategy. But maybe I can move this responsibility to TemplateTypeFactory, actually.